File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/submodel/metamodel/map/qualifier/qualifiable/Constraint.java
package org.eclipse.basyx.submodel.metamodel.map.qualifier.qualifiable;

import org.eclipse.basyx.submodel.metamodel.api.qualifier.qualifiable.IConstraint;
import org.eclipse.basyx.submodel.metamodel.map.modeltype.ModelType;
import org.eclipse.basyx.vab.model.VABModelMap;

public abstract class Constraint extends VABModelMap<Object> implements IConstraint {
	public static final String MODELTYPE = "Constraint";

	public Constraint() {
		// Add model type
		putAll(new ModelType(MODELTYPE));

	}
}

File: sdks/c++/basys.sdk.cc/src/submodel/submodel/map/qualifier/qualifiable/Constraint.cpp
/*
 * Constraint.cpp
 *
 *      Author: wendel
 */

#include <BaSyx/submodel/map/qualifier/qualifiable/Constraint.h>

#include <BaSyx/submodel/map/modeltype/ModelType.h>

namespace basyx {
namespace submodel {

Constraint::Constraint()
  : vab::ElementMap {ModelType(std::string(Path::ModelType)).getMap()}
{}

Constraint::Constraint(basyx::object object)
  : vab::ElementMap {object}
{}

Constraint::Constraint(const IConstraint & constraint)
  : vab::ElementMap {ModelType(std::string(Path::ModelType)).getMap()}
{}

}
}
